packages/core/src/lib/light_push/light_push.ts

```diff
@@ -135,27 +209,95 @@ export class LightPushCore {
       bytes.append(chunk);
     });
 
+    if (bytes.length === 0) {
+      return {
+        success: null,
+        failure: {
+          error: LightPushError.NO_RESPONSE,
+          peerId: peerId
+        }
+      };
+    }
+
+    // Handle response based on protocol version
+    if (protocol === LightPushCodecV3) {
+      return this.handleV3Response(bytes, peerId);
+    } else {
+      return this.handleV2Response(bytes, peerId);
+    }
+  }
+
+  private handleV3Response(
+    bytes: Uint8ArrayList,
+    peerId: PeerId
+  ): LightPushCoreResult {
+    try {
+      const decodedRpcV3 = PushRpcV3.decodeResponse(bytes);
+      const statusCode = decodedRpcV3.statusCode;
+      const statusDesc = decodedRpcV3.statusDesc;
+
+      if (!isV3Success(statusCode)) {
+        // For backward compatibility, map all v3 non-success status codes to REMOTE_PEER_REJECTED
+        // while still propagating statusCode & statusDesc for advanced users
```

Comment:
Why? Do anyone use `REMOTE_PEER_REJECTED` error so you feel like we need to keep the backward compatibility?